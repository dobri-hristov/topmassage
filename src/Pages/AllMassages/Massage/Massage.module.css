.container {
    display: flex;
    flex-direction: column;
    width: 70%;
    min-height: 270px;
    margin: 20px;
    box-shadow: 0px 0px 4px 0px rgba(214, 206, 214, 1);
    background-color: hsl(0, 50%, 98%);
}


.title {
    padding: 10px;
    margin-top: 0;
    margin-bottom: 10px;
    color: rgb(42, 110, 42);
    background-color: #DFE4C9;
}

.imgInfoContainer {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}


.imgContainer {
    height: 200px;
    width: 400px;
}

.imgContainer img {
    width: 100%;
    height: 100%;
    object-fit: cover;
}

.infoContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    flex: 50%;
    margin: 0px 5px 0 30px;
}

.infoContainer {
    margin: 0px 10px 0 10px;
}

.link {
    text-decoration: none;
    font-weight: 500;
    float: right;
    color: black;
    margin: 10px 40px;
}

.link:hover {
    font-weight: 800;
    color: rgb(42, 110, 42);
}

.link i {
    margin-right: 5px;
}

@media screen and (max-width: 1250px) {
    .container {
        margin: 0 auto;
        margin-bottom: 40px;
    }

    .infoContainer {
        margin: 0px 10px 0 10px;
    }

    .imgContainer {
        height: 300px;
        width: 100%;
    }

    .title {
        margin-bottom: 0px;
    }
}

@media screen and (max-width: 889px) {
    .container {
        width: 95%;
        margin-bottom: 40px;
    }

    .headline {
        display: block;
        text-align: center;
    }

    .imgContainer {
        height: 350px;
        width: 100%;
    }
}