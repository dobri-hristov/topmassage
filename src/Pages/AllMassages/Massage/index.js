import React from 'react'
import { Link } from 'react-router-dom'
import styles from './Massage.module.css'

const Massage = ({ data }) => {
    return (
        <div className={styles.container} >
            <h3 className={styles.title}>{data.massageName}</h3>
            <div className={styles.imgInfoContainer}>
                <div className={styles.imgContainer}>
                    <img src={data.img} alt={data.massageName} />
                </div>
                <div className={styles.infoContainer}>
                    <p>{data.massageInfo}</p>
                    <div>
                        <Link className={styles.link} to={`/massages/${data.path}`}>
                            <i className="fas fa-search"></i>
                            Вижте повече!
                        </Link>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Massage
