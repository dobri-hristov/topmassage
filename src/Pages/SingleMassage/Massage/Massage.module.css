.path {
    display: none;
    margin-left: 30px;
    color: #757594
}

.container {
    width: 70%;
    box-shadow: 0px 0px 5px 0px rgb(0 0 0 / 75%);
    margin: 0 auto;
    border-radius: 10px;
    background-color: white;
}

.info {
    padding: 20px;
    line-height: 22px;
    font-size: 18px;
    font-weight: 400;
}

.title {
    text-align: center;
    padding: 10px;
    margin-top: 0;
    margin-bottom: 0;
    color: rgb(42, 110, 42);
    background-color: #DFE4C9;
}

.firstImg {
    width: 100%;
    height: 400px;
}

.secondImg {
    width: 50%;
    height: 300px;
    float: left;
    margin: 0 15px 5px 0;
}

.firstImg img,
.secondImg img {
    width: 100%;
    height: 100%;
    object-fit: cover;
}

.secondImg img {
    border-radius: 10px;
}

.priceContainer {
    display: flex;
    align-items: center;
}

.prices {
    display: flex;
}

.price {
    margin: 10px;
    background-color: #757594;
    color: white;
    border-radius: 20px;
    padding: 5px 10px;
}

@media screen and (max-width: 950px) {
    .path {
        display: block;
    }

    .container {
        width: 95%;
        text-align: center;
    }

    .secondImg {
        margin: 0 auto;
        width: 100%;
        height: 300px;
        float: none;
        margin-top: 18px;
    }
}

@media screen and (max-width: 640px) {
    .priceContainer {
        flex-wrap: wrap;
    }

    .priceContainer h3 {
        flex-basis: 100%;
    }

    .prices {
        margin: 0 auto;
    }
}

@media screen and (max-width: 580px) {
    .prices {
        flex-direction: column;
    }

    .container {
        box-shadow: 0px 0px 4px 0px rgba(214, 206, 214, 1);
        border-radius: 0;
    }
}