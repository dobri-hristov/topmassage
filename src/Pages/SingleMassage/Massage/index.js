import React, { Fragment } from 'react'
import styles from './Massage.module.css'

const Massage = ({ data }) => {
    return (
        <Fragment>
            <div className={styles.path}>
                <h3>Масажи/{data.name}</h3>
            </div>
            <div className={styles.container}>
                <h1 className={styles.title}>{data.name}</h1>
                <div className={styles.firstImg}>
                    <img src={data.mainImg} alt="classic massage" />
                </div>
                <div className={styles.info}>
                    <div>{data.mainInfo}</div>
                    <div>
                        {
                            data.secondImg
                                ? <div className={styles.secondImg}>
                                    <img src={data.secondImg} alt="classic massage" />
                                </div>
                                : null
                        }
                        {
                            data.moreInfo.map((info, index) => (<p key={index}>{info}</p>))
                        }
                    </div>
                    <div className={styles.priceContainer}>
                        <h3>Цени:</h3>
                        <div className={styles.prices}>
                        {
                            Object.keys(data.prices).map((key, index) => (
                                <p className={styles.price} key={index}>{key} мин. - {data.prices[key]} лв. </p>
                            ))
                        }
                        </div>
                    </div>
                </div>
            </div>
        </Fragment>
    )
}

export default Massage
