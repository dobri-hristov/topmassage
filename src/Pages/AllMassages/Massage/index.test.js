import React from 'react'
import TestRenderer from 'react-test-renderer'
import Massage from './index.js'

jest.mock('react-router-dom', () => ({
    Link: 'Link'
}))

describe('Massages page/Massage component', () => {
    test('should render correctly with data', () => {
        const mydata = {
            massageName: 'Класически',
            img: '',
            massageInfo: 'класически масаж',
            path: 'classic'
        }
        const component = TestRenderer.create(
            <Massage data={mydata} />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})