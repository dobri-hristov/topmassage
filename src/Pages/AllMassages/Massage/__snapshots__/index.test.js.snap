// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Massages page/Massage component should render correctly with data 1`] = `
<div
  className="container"
>
  <h3
    className="title"
  >
    Класически
  </h3>
  <div
    className="imgInfoContainer"
  >
    <div
      className="imgContainer"
    >
      <img
        alt="Класически"
        src=""
      />
    </div>
    <div
      className="infoContainer"
    >
      <p>
        класически масаж
      </p>
      <div>
        <Link
          className="link"
          to="/massages/classic"
        >
          <i
            className="fas fa-search"
          />
          Вижте повече!
        </Link>
      </div>
    </div>
  </div>
</div>
`;
